File: PHP/pop3.php
<?php
  $arrMails = array();
  $strResult = 'n/a';
  
  $objPop3 = new COM ("AxEmail.Pop3", NULL, CP_UTF8);
      
  // A license key is required to unlock this component after the trial period has expired.
  // Assign the LicenseKey property every time a new instance of this component is created (see code below). 
  // Alternatively, the LicenseKey property can be set automatically. This requires the license key to be stored in the registry.
  // For details, see manual, chapter "Product Activation".
  //
  // $objPop3->LicenseKey = "XXXXX-XXXXX-XXXXX"; 

  $objPop3->LogFile = sys_get_temp_dir()."Pop3.log"; 
  // Windows default: 'C:\Windows\Temp\Pop3.log'
  
  if (isset($_POST['btnListMessages']))
  {
    $objEmailConstants = new COM("AxEmail.Constants", NULL, CP_UTF8);
    $objEmail = new COM("AxEmail.Message", NULL, CP_UTF8);
    
    $objPop3->Authentication = $objEmailConstants->POP3_AUTH_AUTO;
    
    if (isset($_POST['cbxSecure']))
      $objPop3->SetSecure(995); // 995 is the default secure POP3 port
    
    if ($objPop3->LastError == 0)
    {
      $objPop3->Connect($_POST['txtHost'], $_POST['txtAccount'], $_POST['txtPassword']);
    }
    
    if ($objPop3->LastError == 0)
    {
      $numMessages = $objPop3->CountMessages();
    }
    
    if ($objPop3->LastError == 0)
    {
      for ($i = 1; $i <= $numMessages; $i++)
      {
        $objEmail = $objPop3->GetEmailHeader($i);
        
        
        if ($objPop3->LastError == 0)
          $arrMails[] = $objEmail;
      }
    }
    
    $strResult = $objPop3->LastError . ": " . $objPop3->GetErrorDescription($objPop3->LastError);    
  }

// HTML-CSS layout includes, no code there!
include('css/Header.html'); 
include('css/Menu.html'); 
?>
    <div class="container">
      <h1>Auron Email Component PHP POP3 Sample</h1>
      <hr />
      <p>
        Simple Mail Transfer Protocol (SMTP) is an Internet standard for electronic mail (e-mail) 
        transmission across Internet Protocol (IP) networks.
        SMTP is specified for outgoing mail transport and uses TCP port 25 by default.
      </p>
      <form action="pop3.php" method="post">
        <h2>Component:</h2>
        <h3>Build: <?php echo $objPop3->Build ?>; Module: <?php echo $objPop3->Module ?></h3>
      
        <!-- Host, Secure -->
        <label for="Host">Mailserver:</label>
        <p>
          <input type="text" id="Host" name="txtHost" value="pop3.mycompany.com" />
          
          <input type="checkbox" class="cbFix" id="Secure" name="cbxSecure" checked />
          <label for="Secure">Secure</label>
        </p>
        
        <!-- Account -->
        <label for="Account">Account:</label>
        <p>
          <input type="text" id="Account" name="txtAccount" value="user@mycompany.com" />
        </p>
        
        <!-- Password -->
        <label for="Password">Password:</label>
        <p>
          <input type="password" id="Password" name="txtPassword" value="" />
        </p>
        
        <!-- Empty row -->
        <div class="clearRow"></div>
        
        <!-- List Messages button -->
        <div class="clearLabel"></div>
        <p>
          <input type="submit" name="btnListMessages" value="List Messages" />
        </p>
        
        <!-- Messages Listbox -->
        <label for="Received">Messages received:</label>
        <p>
          <select id="Received" class="FullWidth" name="lvMessages" size="10">
          <?php
          foreach($arrMails as $Mail)
          {
            echo '
            <option>'.$Mail->Date . ": " . $Mail->FromAddress . "; Subject " . $Mail->Subject.'</option>';
          }
          ?>
          </select>
        </p>
        
        <!-- Result -->
        <label for="Result" class="Bold">Result:</label>
        <p>
          <input type="text" id="Result" name="txtResult" class="FullWidth Bold" class="FullWidth" value="<?php echo $strResult; ?>" />
        </p>
      </form>  
      <p>
        This demo uses the Auron Email Component, an <a href="https://www.auronsoftware.com" target="_blank">Auron Software</a> product.<br />
        <a href="index.php">Back to main page</a> 
      </p>
    </div><!-- /container -->
<?php 
//HTML-CSS layout include, no code there!
include('css/Footer.html'); ?>

File: Visual C++/Console Applications/Pop3/Pop3.cpp
#include <comdef.h>
#include <atlbase.h>
#include <windows.h>
#include <stdio.h>

////////////////////////////////////////////////////////////////////////////////

#import "..\..\include\AxEmail.tlb"
#include "..\..\include\AxEmailConstants.h"

////////////////////////////////////////////////////////////////////////////////

LPTSTR ReadInput( LPCTSTR lptszTitle, BOOL bAllowEmpty = FALSE );

////////////////////////////////////////////////////////////////////////////////


int main(void)
{

	LONG			lLastError		= 0L;
	LONG			lNumMessages	= 0L;
	INT				i;
    _bstr_t			bstrHostName, bstrAccount, bstrPassword;
	LPTSTR			cp;
	BOOL			bSecure			= FALSE;
	AxEmail::IPop3Ptr oPop3 = NULL;
	AxEmail::IMessagePtr oMessage = NULL;


    // initialize COM
	CoInitialize(NULL);

	oPop3.CreateInstance(__uuidof(AxEmail::Pop3));
	if( oPop3 == NULL )
	{
		_tprintf( _T("Failed to create AxEmail:Pop3.\n") );		
		goto _EndMain;
	}

	// A license key is required to unlock this component after the trial period has expired.
	// Assign the LicenseKey property every time a new instance of this component is created (see code below). 
	// Alternatively, the LicenseKey property can be set automatically. This requires the license key to be stored in the registry.
	// For details, see manual, chapter "Product Activation".
	/*
	oPop3->LicenseKey = _T("XXXXX-XXXXX-XXXXX");
	*/
  
  // Component info
	_tprintf( _T("Auron Email Component %s\nBuild: %s\nModule: %s\nLicense Status: %s\nLicense Key: %s\n\n"), ( LPCTSTR ) oPop3->Version, ( LPCTSTR ) oPop3->Build, ( LPCTSTR ) oPop3->Module, ( LPCTSTR ) oPop3->LicenseStatus, ( LPCTSTR ) oPop3->LicenseKey );

  // Log File
  // oPop3->LogFile = _T("C:\\Pop3.log");
	_tprintf( _T("Log File used: %s\n\n"), ( LPCTSTR ) oPop3->LogFile );

  
	cp				= ReadInput( _T("POP3 Server") );
	bSecure			= _tcsicmp( cp, _T("pop.gmail.com") ) == 0;
	bstrHostName	= cp;
	bstrAccount		= ReadInput( _T("POP3 mailbox account") );
	bstrPassword	= ReadInput( _T("POP3 mailbox password") );

	if( bSecure )
	{
		oPop3->SetSecure( 995 );
		_tprintf( _T("SetSecure, result: %ld (%s)\n"), oPop3->LastError, ( LPCTSTR ) oPop3->GetErrorDescription( oPop3->LastError ) ); 
	}

	// Connect
	_tprintf( _T("Connecting...\n") );
	oPop3->Connect ( bstrHostName, bstrAccount, bstrPassword );
	_tprintf( _T("Connect, result: %ld (%s)\n"), oPop3->LastError, ( LPCTSTR ) oPop3->GetErrorDescription( oPop3->LastError ) ); 
	if( oPop3->LastError != 0 )
		goto _EndMain;

	// Count messages
	lNumMessages = oPop3->CountMessages();
	_tprintf( _T("CountMessages, result: %ld (%s)\n"), oPop3->LastError, ( LPCTSTR ) oPop3->GetErrorDescription( oPop3->LastError ) ); 
	if( lLastError != 0 )
		goto _EndMain;

	_tprintf( _T("Number of messages in mailbox: %ld\n"), lNumMessages );

	for(  i = 1; i < lNumMessages + 1 ; i++ )
	{
		oMessage = oPop3->GetEmailHeader( i ); 

		if( oMessage == NULL )
			continue;

		_tprintf( _T("Mail ID       : %d\n"), oMessage->ID );
		_tprintf( _T("  Mail From   : %s\n"), ( LPCTSTR ) oMessage->FromAddress );	
		_tprintf( _T("  Mail To     : %s\n"), ( LPCTSTR ) oMessage->ToAddress );	
		_tprintf( _T("  Mail Subject: %s\n"), ( LPCTSTR ) oMessage->Subject );	
		_tprintf( _T("  Mail Size   : %ld\n"), oMessage->Size );
		_tprintf( _T("  Mail Sent   : %s\n"), ( LPCTSTR ) oMessage->Date );
			
		_tprintf( _T("\n") );
	}

	oPop3->Disconnect();

_EndMain:

	if( oPop3 != NULL )
	{
		oPop3.Release();
	}

	_tprintf( _T("Ready.\n") );

	return 0;
}

////////////////////////////////////////////////////////////////////////////////

LPTSTR ReadInput( LPCTSTR lptszTitle, BOOL bAllowEmpty )
{
	static TCHAR		tszInput [ 255 + 1 ] = { _T('\0') };

	_tprintf( _T("%s:\n"), lptszTitle );
	do
	{
		_tprintf ( _T("   > ") );
		// scanf ( "%s", tszInput );
		fflush(stdin); 
		fflush(stdout); 
		_fgetts( tszInput, 255, stdin );
		if( tszInput[ 0 ] != _T('\0') && tszInput[ _tcsclen( tszInput ) - 1  ] == _T('\n') )
			tszInput[ _tcsclen( tszInput ) - 1  ] = _T('\0');
	} while( _tcsclen ( tszInput ) == 0 && ! bAllowEmpty );

	return tszInput;
}

///////////////////////////////////////////////////////////////////////////////////////////


